src/k8s/pkg/snap/snap_test.go

```diff
@@ -0,0 +1,63 @@
+package snap_test
+
+import (
+	"context"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/canonical/k8s/pkg/snap"
+)
+
+func TestServiceStartStop(t *testing.T) {
+	mockRunner := &MockRunner{}
+	s := snap.NewSnap("testdir", "testdir", "testdir", snap.WithCommandRunner(mockRunner.Run))
+
+	tests := []struct {
+		name            string
+		action          func(ctx context.Context, service string) error
+		service         string
+		expectedCommand string
+	}{
+		{
+			name:            "StartService",
+			action:          s.StartService,
+			service:         "test-service",
+			expectedCommand: "snapctl start --enable k8s.test-service",
+		},
+		{
+			name:            "StopService",
+			action:          s.StopService,
+			service:         "test-service",
+			expectedCommand: "snapctl stop --disable k8s.test-service",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockRunner.CalledWithCommand = []string{} // Resetting the commands for each test case
+			tc.action(context.Background(), tc.service)
+			if lastCmd := mockRunner.CalledWithCommand[0]; lastCmd != tc.expectedCommand {
+				t.Fatalf("Expected command %q, but %q was called instead for service %s", tc.expectedCommand, lastCmd, tc.service)
+			}
+		})
+	}
+}
```

Comment:
Some notes on the test added:

1. This will produce tests with names like:

```
TestServiceStartStop/StartService
TestServiceStartStop/StopService
```

which stutters. It would be easier if you just call it `TestServices`

2. It would probably be easier if you created a separate snap instance inside each sub-test, that way you don't need the reset step or accidentally sharing state

3. Use `gomega` instead, see other places.

4. No real need for a table here. It should be simpler (and shorter) to just write the tests explicitly.

Taking the above into account, consider the following:

```go

func TestServices(t *testing.T) {
	t.Run("Start", func(t *testing.T) {
		g := NewWithT(t)
		mockRunner := &MockRunner{}
		snap := snap.NewSnap("testdir", "testdir", snap.WithCommandRunner(mockRunner.Run))

		err := snap.StartService(context.Background(), "service")
		g.Expect(err).To(BeNil())
		g.Expect(mockRunner.CalledWithCommand).To(ConsistOf("snapctl start --enable k8s.test-service"))

		t.Run("Fail", func(t *testing.T) {
			g := NewWithT(t)
			mockRunner.Err = fmt.Errorf("some error")

			err := snap.StartService(context.Background(), "service")
			g.Expect(err).NotTo(BeNil())
		})
	}

	t.Run("Stop", func(t *testing.T) {
		mockRunner := &MockRunner{}
		snap := snap.NewSnap("testdir", "testdir", snap.WithCommandRunner(mockRunner.Run))

		err := snap.StopService(context.Background(), "service")
		g.Expect(err).To(BeNil())
		g.Expect(mockRunner.CalledWithCommand).To(ConsistOf("snapctl stop --disable k8s.test-service"))

		t.Run("Fail", func(t *testing.T) {
			g := NewWithT(t)
			mockRunner.Err = fmt.Errorf("some error")

			err := snap.StartService(context.Background(), "service")
			g.Expect(err).NotTo(BeNil())
		})
	})
}
```

Reply:
use the above as inspiration, let's also have unit tests for what happens when the command has failed